### ISSUE TITLE
Make server multi-tenant

### ISSUE THREAD

**Issue by mikehelmick** [google/exposure-notifications-verification-server#104]
### TL;DR

Make the server multi-tenant, allowing for different "realms".

### Design

**Proposal**

All realm type.
A user can be authorized to use a realm and or authorized to be an admin for a realm.

API keys will be scoped to an individual realm.

Certificate keys will be on a per-realm basis.

Verification codes - open question on if these should be a large shared namespace. if API Keys are realm specific, then this could be done.

**Alternatives considered**

* large single tenant installation - means certificates can potentially cross PHA authority.
* multi-single-tenant installations, more operationally burdensome.

/kind enhancement
